src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -111,14 +112,46 @@ private static void checkHasNaturalAlignment(FunctionDescriptor descriptor) {
     private static void checkHasNaturalAlignmentRecursive(MemoryLayout layout) {
         checkHasNaturalAlignment(layout);
         if (layout instanceof GroupLayout gl) {
+            boolean isCheckRequired = true;
             for (MemoryLayout member : gl.memberLayouts()) {
-                checkHasNaturalAlignmentRecursive(member);
+                isCheckRequired = isAlignmentCheckRequired(member, isCheckRequired);
+                if (isCheckRequired) {
+                    checkHasNaturalAlignmentRecursive(member);
+                }
             }
         } else if (layout instanceof SequenceLayout sl) {
             checkHasNaturalAlignmentRecursive(sl.elementLayout());
         }
     }
 
+    private static boolean isAlignmentCheckRequired(MemoryLayout layout, boolean isCheckRequired) {
+        boolean isAixOS = System.getProperty("os.name").toLowerCase().contains("aix");
```

Comment:
`isAixOS` should be a static field so `os.name` need be checked at most once and should just accept "AIX" (see [ProcessImpl](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/src/java.base/unix/classes/java/lang/ProcessImpl.java#L146))

Reply:
Fixed as suggested above.